core/kazoo_bindings/src/kazoo_bindings.erl

```diff
@@ -710,7 +713,9 @@ map_processor_fold(#kz_binding{binding_parts=BParts
                   ,_Routing
                   ,RoutingParts
                   ) ->
-    case matches(BParts, RoutingParts) of
+    case matches(BParts, RoutingParts)
+        orelse matches(RoutingParts, BParts)
```

Comment:
This is due to the fact that kazoo_bindings:matches/2 ignores *s and #s (their special meaning anyway) in the second argument. That mean kazoo_bindings:map/2 & fold/2 's first argument isn't properly matched.
matches/2 only interprets these chars as special in its first argument.

There are no test cases per se but the tasks_bindings:apply/3 code.